templates/kops/kops-private-topology.yaml.gotmpl

```diff
@@ -201,24 +199,42 @@ spec:
   kubernetesVersion: {{ getenv "KUBERNETES_VERSION" }}
   masterPublicName: api.{{ getenv "KOPS_CLUSTER_NAME" }}
   networkCIDR: {{ getenv "KOPS_NETWORK_CIDR" "172.20.0.0/16" }}
+  {{- if getenv "KOPS_CONFIGURED_VPC_ID" }}
```

Comment:
Generally we've followed the convention of using the canonical names rather than mappping them to new ones.

Reply:
Generally I would agree with that, but I feel this is a special case worthy of deviation. This field is not normally specified, and when it is, it is a VPC ID.  I tried several variants of searching for kops vpc ID versus network ID and always got more (and more relevant) results with VPC ID. Also, VPC ID is the canonical name for the value in the rest of our architecture and in AWS.  So in this case I think calling it `KOPS_CONFIGURED_NETWORK_ID` would be more confusing than helpful.

Reply:
See, for example, our own `kops-metadata` [module's output](https://github.com/cloudposse/terraform-aws-kops-metadata/blob/cea5cba634b2659aefb6a2e6821161d7081baa1d/outputs.tf#L1-L5)

Reply:
See also the `kops` [documentation for running in an exiting VPC](https://github.com/kubernetes/kops/blob/master/docs/run_in_existing_vpc.md#running-in-a-shared-vpc)

>1. Use `kops create cluster` with the `--vpc` argument for your existing VPC:

  ```shell
  export KOPS_STATE_STORE=s3://<somes3bucket>
  export CLUSTER_NAME=<sharedvpc.mydomain.com>
  export VPC_ID=vpc-12345678 # replace with your VPC id
  export NETWORK_CIDR=10.100.0.0/16 # replace with the cidr for the VPC ${VPC_ID}

  kops create cluster --zones=us-east-1b --name=${CLUSTER_NAME} --vpc=${VPC_ID}
  ```

>2. Then `kops edit cluster ${CLUSTER_NAME}` will show you something like:

  ```yaml
  metadata:
    creationTimestamp: "2016-06-27T14:23:34Z"
    name: ${CLUSTER_NAME}
  spec:
    cloudProvider: aws
    networkCIDR: ${NETWORK_CIDR}
    networkID: ${VPC_ID}
    nonMasqueradeCIDR: 100.64.0.0/10
    subnets:
    - cidr: 172.20.32.0/19
      name: us-east-1b
      type: Public
      zone: us-east-1b
  ```

Reply:
What about `s/CONFIGURED/EXISTING/`

Reply:
I find `EXISTING` problematic despite its use in the documentation. How about `SHARED`, which matches the kubernetes terminology of "owned" vs "shared".

Reply:
Shared sounds good